### ISSUE TITLE
Number inputs in RTL have the minus sign in the wrong place while typing.

### ISSUE THREAD

**Issue by rachel-fenichel** [google/blockly#1802]
![443a36d2-b2a9-4ad6-a647-2900eeda99e9](https://user-images.githubusercontent.com/13686399/38906309-b3ef0ff2-426a-11e8-96b4-d92a04fed7a5.gif)

We can fix this by putting dir="ltr" on the widget div when it's a number field (even in rtl), becuase numbers should just be ltr.

**Comment by AnmAtAnm** [comment]
This is a locale specific issue.  In some locales, this is correct: https://docs.microsoft.com/en-us/globalization/locale/number-formatting

**Comment by chrisgarrity** [comment]
For Hebrew, numbers, and math in general, are always LTR. I haven't checked with anyone regarding Arabic. This is likely to be a locale by locale issue rather than just RTL vs LTR.

**Comment by AnmAtAnm** [comment]
I believe Neil's decision (for simplicity, right or not) on this is to always leave number rendering in LTR. However, sometime the OS has different choices and this comes out in the field pop-up widgets, like number fields.

**Comment by NeilFraser** [comment]
This is not intentional at all, and I'm pretty sure this is a regression.  There's code to avoid this here:
https://github.com/google/blockly/blob/master/core/widgetdiv.js#L87

**Comment by NeilFraser** [comment]
Oh, I see.  The code mentioned above is what's *causing* this issue.  But remove it, and non-numbers would break.  Lovely.

**Comment by rachel-fenichel** [comment]
Right, numbers and non-numbers really just need to behave differently here.  Chris just added some code in scratch blocks to handle this: https://github.com/LLK/scratch-blocks/pull/1702/files#diff-393b262d7ed6406aa328f2e57e089e5fR607

**Comment by NeilFraser** [comment]
@chrisgarrity Math in general is LTR in Hebrew.  However in Arabic and related languages it may be LTR or (more commonly) RTL depending on local culture (not language or even country).  Thus "2 - 4" is completely ambiguous.

**Comment by NeilFraser** [comment]
@rachel-fenichel That's a very ugly hack; an exception in field.js specific to the math_number block.  It should be in Blockly.FieldNumber since there are a lot of other places with numeric input.
